packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/app_lifecycle_provider.dart

```diff
@@ -0,0 +1,17 @@
+abstract class AppLifecycleProvider {
+  bool _isObserving = false;
+
+  AppLifecycleProvider();
+
+  void setOnForegroundListener(Function onForeground);
+  void setOnBackgroundListener(Function onBackground);
+
+  void startObserving() {
```

Comment:
You might want to call `@mustCallSuper` here to enforce that these are called. Either that or just remove the behavior of `startObserving` & `stopObserving`.

 You re-implemented the `_isObserving` logic in `FlutterAppLifecycleProvider` instead of calling super. If this behavior is specific to `FlutterAppLifecycleProvider` then remove it from this class. If it should be part of all AppLifecycleProvider implementations, then add `@mustCallSuper` and call super from `FlutterAppLifecycleProvider`.

Reply:
Thanks removed the abstract class method implementations.